src/lib/logical_query_plan/lqp_utils.cpp

```diff
@@ -231,7 +231,11 @@ std::set<std::string> lqp_find_modified_tables(const std::shared_ptr<AbstractLQP
   return modified_tables;
 }
 
-std::shared_ptr<AbstractExpression> lqp_subplan_to_boolean_expression(const std::shared_ptr<AbstractLQPNode>& lqp) {
+std::shared_ptr<AbstractExpression> lqp_subplan_to_boolean_expression(
+    const std::shared_ptr<AbstractLQPNode>& lqp,
+    const std::function<bool(const std::shared_ptr<AbstractLQPNode>& lqp)>& node_is_allowed) {
```

Comment:
Na, adding the std::function as a parameter is quite atrocious. The JitAwareLQPTranslator is the only user of this function - either make `base_node` a parameter or copy the function to the JIT and do with it whatever you want. But a general-purpose freestanding function shouldn't use an obscure std::function parameter hack.

Reply:
I added a parameter `base_node`.